be/src/pipeline/dependency.h

```diff
@@ -84,24 +83,27 @@ struct BasicSharedState {
     Dependency* create_sink_dependency(int dest_id, int node_id, std::string name);
 };
 
+/**
+ * We divide dependency into three categories:
+ * 1. BEFORE_EXECUTION: pipeline tasks need to wait for this before execution. (i.e. ExecutionDependency, RuntimeFilterDependency)
+ * 2. DURING_EXECUTION: In execution phase, tasks should be running only if this dependency is ready. (i.e. Read/Write Dependency)
+ * 3. AFTER_EXECUTION: After execution phase, tasks should be closed after this dependency is ready.
+ */
+enum class DependencyType : uint8_t {
```

Comment:
We should not add dependency type.
It is very difficult for the developer to use. Many developer will not know the type related with wake up method.